rhino/src/main/java/org/mozilla/javascript/regexp/NativeRegExp.java

```diff
@@ -1359,6 +1656,22 @@ private static boolean flatNMatcher(
         return true;
     }
 
+    private static boolean flatNMatcherBackward(
+            REGlobalData gData, int matchChars, int length, String input) {
+        if ((gData.cp - length) < 0) return false;
+
+        // in the input, start from cp - 1 and go back length chars
+        // in the regex source, do it the other way
+        for (int i = 1; i <= length; i++) {
+            if (gData.regexp.source[matchChars + length - i] != input.charAt(gData.cp - i)) {
+                return false;
+            }
+        }
+
+        gData.cp -= length;
+        return true;
```

Comment:
It doesn't look like we ever reach this line in tests

Reply:
This is a good find. I have memory of debugging this code with a test262 test so I was not sure why this is happening. Long story short, I now know why this isn't reached.

I've "fixed" it by moving the FLAT concatenation code from[ emitBytecode stage](https://github.com/mozilla/rhino/pull/1843/commits/4e7d0f4e6345764368234e311f02b4ac719d7d52#diff-26b8d191be9a315419e6621d92ee9a0c274c8bd7f408a2d7220bce18c80b97a7L1448) to the [parsing stage](https://github.com/mozilla/rhino/pull/1843/commits/4e7d0f4e6345764368234e311f02b4ac719d7d52#diff-26b8d191be9a315419e6621d92ee9a0c274c8bd7f408a2d7220bce18c80b97a7R792). This also makes lookbehinds a bit more optimal.